### ISSUE TITLE
Ubuntu WebGL Build Error

### ISSUE THREAD

**Issue by josephdigipen** [game-ci/unity-builder#205]
**Bug description**

New project with one scene. No custom builds step.
Builds locally in WebGL.
Error in CI using v2.0-alpha-6

**How to reproduce**

runs-on: ubuntu-latest
        
uses: game-ci/unity-builder@v2.0-alpha-6
        with:
          targetPlatform: WebGL
          unityVersion: 2020.1.1f1

Errors Out:
```
2021-01-25T19:40:50.6235448Z 
2021-01-25T19:40:50.6236214Z stdout:
2021-01-25T19:40:50.6237137Z Running il2cpp.exe in server GC mode.
2021-01-25T19:40:50.6238261Z Building build.bc with EmscriptenToolChain
2021-01-25T19:40:50.6239512Z 	Output directory: /github/workspace/Temp/StagingArea/Data/Native
2021-01-25T19:40:50.6240779Z 	Cache directory: /github/workspace/Library/il2cpp_cache
2021-01-25T19:40:50.6242934Z il2cpp.exe didn't catch exception: System.AggregateException: One or more errors occurred. (No such file or directory) (No such file or directory)
2021-01-25T19:40:50.6245157Z  ---> System.ComponentModel.Win32Exception (2): No such file or directory
2021-01-25T19:40:50.6248278Z    at System.Diagnostics.Process.ForkAndExecProcess(String filename, String[] argv, String[] envp, String cwd, Boolean redirectStdin, Boolean redirectStdout, Boolean redirectStderr, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec)
2021-01-25T19:40:50.6251764Z    at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
2021-01-25T19:40:50.6253395Z    at System.Diagnostics.Process.Start()
2021-01-25T19:40:50.6255364Z    at Unity.IL2CPP.Shell.SetupAndStart(ExecuteContext context, Boolean asyncMode) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Shell/Shell.cs:line 329
2021-01-25T19:40:50.6258018Z    at Unity.IL2CPP.Shell.ExecuteAsync(ExecuteArgs executeArgs, IExecuteController controller, Boolean asyncMode) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Shell/Shell.cs:line 256
2021-01-25T19:40:50.6260708Z    at Unity.IL2CPP.Shell.Execute(ExecuteArgs executeArgs, IExecuteController controller) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Shell/Shell.cs:line 277
2021-01-25T19:40:50.6264101Z    at Unity.IL2CPP.Building.CppProgramBuilder.ProvideObjectFile(IntermediateObjectFileCompilationData data) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Building/CppProgramBuilder.cs:line 319
2021-01-25T19:40:50.6267657Z    at Unity.IL2CPP.Building.ParallelFor.<>c__DisplayClass1_0`2.<RunWithResult>b__0(Object o) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Building/ParallelFor.cs:line 73
2021-01-25T19:40:50.6269797Z    --- End of inner exception stack trace ---
2021-01-25T19:40:50.6271846Z    at Unity.IL2CPP.Building.ParallelFor.RunWithResult[D,T](D[] data, Func`2 action) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Building/ParallelFor.cs:line 95
2021-01-25T19:40:50.6275560Z    at Unity.IL2CPP.Building.CppProgramBuilder.BuildAllCppFiles(IEnumerable`1 sourceFilesToCompile, IBuildStatisticsCollector statisticsCollector) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Building/CppProgramBuilder.cs:line 240
2021-01-25T19:40:50.6279401Z    at Unity.IL2CPP.Building.CppProgramBuilder.Build(IBuildStatistics& statistics) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Building/CppProgramBuilder.cs:line 135
2021-01-25T19:40:50.6282561Z    at il2cpp.Compilation.CompilationDriver.Run(RuntimePlatform platform, BuildingOptions buildingOptions) in /Users/bokken/build/output/unity/il2cpp/il2cpp/Compilation/CompilationDriver.cs:line 64
2021-01-25T19:40:50.6285412Z    at il2cpp.Program.DoRun(String[] args, RuntimePlatform platform, BuildingOptions buildingOptions) in /Users/bokken/build/output/unity/il2cpp/il2cpp/Program.cs:line 86
2021-01-25T19:40:50.6287593Z    at il2cpp.Program.Run(String[] args, Boolean setInvariantCulture) in /Users/bokken/build/output/unity/il2cpp/il2cpp/Program.cs:line 71
2021-01-25T19:40:50.6289436Z    at il2cpp.Program.Main(String[] args) in /Users/bokken/build/output/unity/il2cpp/il2cpp/Program.cs:line 36
2021-01-25T19:40:50.6291565Z  ---> (Inner Exception #1) System.ComponentModel.Win32Exception (2): No such file or directory
2021-01-25T19:40:50.6294781Z    at System.Diagnostics.Process.ForkAndExecProcess(String filename, String[] argv, String[] envp, String cwd, Boolean redirectStdin, Boolean redirectStdout, Boolean redirectStderr, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec)
2021-01-25T19:40:50.6297932Z    at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
2021-01-25T19:40:50.6299532Z    at System.Diagnostics.Process.Start()
2021-01-25T19:40:50.6301501Z    at Unity.IL2CPP.Shell.SetupAndStart(ExecuteContext context, Boolean asyncMode) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Shell/Shell.cs:line 329
2021-01-25T19:40:50.6304236Z    at Unity.IL2CPP.Shell.ExecuteAsync(ExecuteArgs executeArgs, IExecuteController controller, Boolean asyncMode) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Shell/Shell.cs:line 256
2021-01-25T19:40:50.6306923Z    at Unity.IL2CPP.Shell.Execute(ExecuteArgs executeArgs, IExecuteController controller) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Shell/Shell.cs:line 277
2021-01-25T19:40:50.6310305Z    at Unity.IL2CPP.Building.CppProgramBuilder.ProvideObjectFile(IntermediateObjectFileCompilationData data) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Building/CppProgramBuilder.cs:line 319
2021-01-25T19:40:50.6314143Z    at Unity.IL2CPP.Building.ParallelFor.<>c__DisplayClass1_0`2.<RunWithResult>b__0(Object o) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Building/ParallelFor.cs:line 73<---
2021-01-25T19:40:50.6315603Z 
2021-01-25T19:40:50.6316310Z stderr:
2021-01-25T19:40:50.6317731Z Unhandled exception. System.AggregateException: One or more errors occurred. (No such file or directory) (No such file or directory)
2021-01-25T19:40:50.6319784Z  ---> System.ComponentModel.Win32Exception (2): No such file or directory
2021-01-25T19:40:50.6322910Z    at System.Diagnostics.Process.ForkAndExecProcess(String filename, String[] argv, String[] envp, String cwd, Boolean redirectStdin, Boolean redirectStdout, Boolean redirectStderr, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec)
2021-01-25T19:40:50.6326073Z    at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
2021-01-25T19:40:50.6327718Z    at System.Diagnostics.Process.Start()
2021-01-25T19:40:50.6329665Z    at Unity.IL2CPP.Shell.SetupAndStart(ExecuteContext context, Boolean asyncMode) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Shell/Shell.cs:line 329
2021-01-25T19:40:50.6332449Z    at Unity.IL2CPP.Shell.ExecuteAsync(ExecuteArgs executeArgs, IExecuteController controller, Boolean asyncMode) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Shell/Shell.cs:line 256
2021-01-25T19:40:50.6335131Z    at Unity.IL2CPP.Shell.Execute(ExecuteArgs executeArgs, IExecuteController controller) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Shell/Shell.cs:line 277
2021-01-25T19:40:50.6338501Z    at Unity.IL2CPP.Building.CppProgramBuilder.ProvideObjectFile(IntermediateObjectFileCompilationData data) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Building/CppProgramBuilder.cs:line 319
2021-01-25T19:40:50.6342000Z    at Unity.IL2CPP.Building.ParallelFor.<>c__DisplayClass1_0`2.<RunWithResult>b__0(Object o) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Building/ParallelFor.cs:line 73
2021-01-25T19:40:50.6344166Z    --- End of inner exception stack trace ---
2021-01-25T19:40:50.6346187Z    at Unity.IL2CPP.Building.ParallelFor.RunWithResult[D,T](D[] data, Func`2 action) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Building/ParallelFor.cs:line 95
2021-01-25T19:40:50.6349915Z    at Unity.IL2CPP.Building.CppProgramBuilder.BuildAllCppFiles(IEnumerable`1 sourceFilesToCompile, IBuildStatisticsCollector statisticsCollector) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Building/CppProgramBuilder.cs:line 240
2021-01-25T19:40:50.6353719Z    at Unity.IL2CPP.Building.CppProgramBuilder.Build(IBuildStatistics& statistics) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Building/CppProgramBuilder.cs:line 135
2021-01-25T19:40:50.6356853Z    at il2cpp.Compilation.CompilationDriver.Run(RuntimePlatform platform, BuildingOptions buildingOptions) in /Users/bokken/build/output/unity/il2cpp/il2cpp/Compilation/CompilationDriver.cs:line 64
2021-01-25T19:40:50.6359597Z    at il2cpp.Program.DoRun(String[] args, RuntimePlatform platform, BuildingOptions buildingOptions) in /Users/bokken/build/output/unity/il2cpp/il2cpp/Program.cs:line 86
2021-01-25T19:40:50.6361780Z    at il2cpp.Program.Run(String[] args, Boolean setInvariantCulture) in /Users/bokken/build/output/unity/il2cpp/il2cpp/Program.cs:line 71
2021-01-25T19:40:50.6363680Z    at il2cpp.Program.Main(String[] args) in /Users/bokken/build/output/unity/il2cpp/il2cpp/Program.cs:line 36
2021-01-25T19:40:50.6365640Z  ---> (Inner Exception #1) System.ComponentModel.Win32Exception (2): No such file or directory
2021-01-25T19:40:50.6368810Z    at System.Diagnostics.Process.ForkAndExecProcess(String filename, String[] argv, String[] envp, String cwd, Boolean redirectStdin, Boolean redirectStdout, Boolean redirectStderr, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec)
2021-01-25T19:40:50.6372119Z    at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
2021-01-25T19:40:50.6373736Z    at System.Diagnostics.Process.Start()
2021-01-25T19:40:50.6375717Z    at Unity.IL2CPP.Shell.SetupAndStart(ExecuteContext context, Boolean asyncMode) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Shell/Shell.cs:line 329
2021-01-25T19:40:50.6378363Z    at Unity.IL2CPP.Shell.ExecuteAsync(ExecuteArgs executeArgs, IExecuteController controller, Boolean asyncMode) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Shell/Shell.cs:line 256
2021-01-25T19:40:50.6381037Z    at Unity.IL2CPP.Shell.Execute(ExecuteArgs executeArgs, IExecuteController controller) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Shell/Shell.cs:line 277
2021-01-25T19:40:50.6384445Z    at Unity.IL2CPP.Building.CppProgramBuilder.ProvideObjectFile(IntermediateObjectFileCompilationData data) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Building/CppProgramBuilder.cs:line 319
2021-01-25T19:40:50.6388231Z    at Unity.IL2CPP.Building.ParallelFor.<>c__DisplayClass1_0`2.<RunWithResult>b__0(Object o) in /Users/bokken/build/output/unity/il2cpp/Unity.IL2CPP.Building/ParallelFor.cs:line 73<---
```

**Expected behavior**

Build

Not sure why so many reference to Win32 or if this is normal to run il2cpp.exe while running on ubuntu.

**Comment by webbertakken** [comment]
WebGL IL2CPP has been fixed in https://github.com/game-ci/docker/pull/61 and will be released in the next image version (0.9). The version of the actions that you have set up will automatically switch to that new version.

We're working hard to make it happen and roll out new versions. Please allow a few days.

**Comment by davidmfinol** [comment]
@josephdigipen Are you still facing this error?

v0.9 is out now, so I believe this should be resolved.

**Comment by zmays** [comment]
I ran a build today for webgl and got the error mentioned in this ticket: https://github.com/game-ci/docker/issues/75

error:
```
Filename: gzip
Arguments: -9 --keep -S ".compressed" "/github/workspace/Temp/StagingArea/Data/Output/Build/WebGL.data.gz"
index: -1
FileNotFoundException: Could not find file "/github/workspace/Temp/StagingArea/Data/Output/Build/WebGL.data.gz.compressed"
  at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) [0x0019e] in <9577ac7a62ef43179789031239ba8798>:0 
```

yml:
```
name: Build project

on:
  push: { branches: [main] }

env:
  UNITY_LICENSE: ${{ secrets.UNITY_LICENSE }}

jobs:
  buildForSomePlatforms:
    name: Build for ${{ matrix.targetPlatform }} on version ${{ matrix.unityVersion }}
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        unityVersion:
          - 2020.1.13f1
        targetPlatform:
          - WebGL
    steps:
      - uses: actions/checkout@v2
        with:
          lfs: true
      - uses: actions/cache@v1.1.0
        with:
          path: ${{ matrix.projectPath }}/Library
          key: Library-${{ matrix.projectPath }}-${{ matrix.targetPlatform }}
          restore-keys: |
            Library-${{ matrix.projectPath }}-
            Library-
      - uses: game-ci/unity-builder@v2.0-alpha-6

        with:
          projectPath: ${{ matrix.projectPath }}
          unityVersion: ${{ matrix.unityVersion }}
          targetPlatform: ${{ matrix.targetPlatform }}
      - uses: actions/upload-artifact@v1
        with:
          name: Build
          path: build

```

**Comment by webbertakken** [comment]
Failed compression has been fixed in https://github.com/game-ci/docker/pull/77 and will be released in the next image version (0.10)The version of the actions that you have set up will automatically switch to that new version.

We're working hard to make it happen and roll out new versions. Please allow a few more days.

**Comment by webbertakken** [comment]
This has been released in https://github.com/game-ci/docker/releases/tag/v0.10. The images should become available soon, and builder v2-alpha will automatically roll to the latest 0.x image.
